runtime/browser/xwalk_browser_context.cc

```diff
@@ -212,6 +213,14 @@ XWalkBrowserContext::GetGuestManager() {
 }
 
 storage::SpecialStoragePolicy* XWalkBrowserContext::GetSpecialStoragePolicy() {
+  LOG(WARNING) << "XWalkBrowserContext::GetSpecialStoragePolicy() :: " <<
+          "--unlimited-storage switch value: " <<
+          (base::CommandLine::ForCurrentProcess()->HasSwitch(
+                  switches::kUnlimitedStorage));
+  // TODO cache the XwalkSpecialStoragePolicy instance?
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kUnlimitedStorage))
+    return new xwalk::XWalkSpecialStoragePolicy();
```

Comment:
I'll have a go at implementing this now, but some questions:

    Should it be initialised lazily?
    If so, does initialisation need to be thread-safe?

For now, I will assume that initialising the `XWalkSpecialStoragePolicy` in the constructor of `XWalkBrowserContext` is appropriate.